doc/cross-platform-development.md

```diff
@@ -0,0 +1,79 @@
+## Introduction
+It's often common practice that a juju developer needs to test out new juju code
+on machines that are not the same operating system or architecture to that of
+their current host. Alternatively in this workflow it is common place to want to
+want to perform Juju upgrades.
+
+Examples of this would be:
+- ubuntu/amd64 -> ubuntu/s390x
+- ubuntu/amd64 -> centos/amd64
+- macos/amd64 -> ubuntu/amd64
+- macos/arm64 -> ubutnu/amd64
+
+Doing this has been difficult in the past but with changes to the juju `Makefile`
+we can now build simple streams for multiple platforms and versions.
+
+## How it works
+### Bootstrapping
+We start by telling the juju build systems that we would like a local simple
+streams repository created for a given set of Go style os/arch pairs. For this
+example we would like a simple stream of our development changes that have
+artifacts for `linux/amd64` and `linux/arm64`.
+
+This is created by running:
+```bash
+AGENT_PACKAGE_PLATFORMS="linux/amd64 linux/arm64" make simplestreams
+```
+This process will build all the juju agent binaries for the platforms specified
+above and package them into a simple streams repository.
+
+The end of the output for this `make` command will also provide the user with an
+`export` statement that the user should run to help the juju bootstrap command
+automatically find this local simple streams repository. Example output:
+```bash
+export JUJU_METADATA_SOURCE="...."
```

Comment:
As per other comment, allow JUJU_METADATA_SOURCE to be already set and use that and omit the export instruction